--- ../src/StandardGovernor.sol	2024-03-11 10:58:22.794901547 +0200
+++ 	2024-03-11 10:59:45.644891672 +0200
@@ -373,7 +373,8 @@
 
         IPowerToken(voteToken).markParticipation(voter_);
 
-        uint256 reward_ = (maxTotalZeroRewardPerActiveEpoch * weight_) / _getTotalSupply(currentEpoch_ - 1);
+        //uint256 reward_ = (maxTotalZeroRewardPerActiveEpoch * weight_) / _getTotalSupply(currentEpoch_ - 1);
+        uint256 reward_ = math.mulDivDown(maxTotalZeroRewardPerActiveEpoch, weight_, _getTotalSupply(currentEpoch_ - 1));
 
         if (reward_ == 0) return;
 
@@ -469,3 +470,9 @@
         return 0;
     }
 }
+
+library math {
+    function mulDivDown(uint256 x, uint256 y, uint256 z) internal pure returns (uint256) {
+        return (x * y) / z;
+    }
+}
